pkgdb/src/buildenv/assets/activate.d/tcsh_home/.tcshrc

```diff
@@ -0,0 +1,21 @@
+# tcsh (unbelievably) does not have a way to source an alternate "rc" file,
+# but it will follow $HOME to a .tcshrc file of our making. This is that
+# file, and we use it to source the files that tcsh would normally source
+# from the user's home directory, after which we can proceed with flox
+# environment initialization.
+
+# Start by reinstating the user's original $HOME.
+setenv HOME "$FLOX_ORIG_HOME"
+
+# By this point we will have already sourced the "system" /etc/*cshrc files,
+# so we can continue with sourcing the user's .login and .{,t}cshrc files.
+foreach f ( .login .cshrc .tcshrc )
+    if ( -f "$HOME/$f" ) then
+	source "$HOME/$f"
+    endif
+end
+
+# Bring in the Nix and Flox environment customizations.
+if ( $?FLOX_TCSH_INIT_SCRIPT ) then
+    source "$FLOX_TCSH_INIT_SCRIPT"
```

Comment:
question blocking: looks like above you source e.g. `$HOME/.cshrc`, but then doesn't `FLOX_TCSH_INIT_SCRIPT` source `~/.cshrc` again? Not sure if I'm reading that right

Reply:
Well-spotted gentlemen! I added a new test to verify that all dotfiles & hooks are specified exactly as expected, in the right order, and only once, and it surfaced that .tcshrc was being sourced twice.  👍 

Patch forthcoming ...

Reply:
commit fa88ed4d688e15b00369b30431981c98644d6022 should do the trick - resolving conversation. 👍